src/app/details/details.component.ts

```diff
@@ -11,16 +13,14 @@ import { ActivatedRoute } from '@angular/router';
 export class DetailsComponent implements OnInit {
 
   userId: number;
-  userData: any;
+  user$: Observable<User>;
 
   constructor(private route: ActivatedRoute, private data: DataService) {
-     this.route.params.subscribe( params => this.userId = params.id );
+    this.route.params.subscribe(params => this.userId = params.id);
   }
 
   ngOnInit() {
-    this.data.getUsersData().subscribe(
-      data => this.userData = data
-    );
+    this.user$ = this.data.getUserData(this.userId);
```

Comment:
We store a reference to the observable so that we don't need to handle the ```.subscribe``` explicitly and we can let angular do that with the ```async``` pipe.